common: modify type specifier enumeration type

jeongmincha opened the pull request:
기존에 VarType으로 typedef되어 있던 enumeration type의 이름을 TypeSpecifier로 변경하였고, STRUCT와 UNION타입을 이 열거형 타입에 추가하였습니다. 이름을 변경한탓에 VarType을 사용하고 있던 소스들을 전부 TypeSpecifier를 사용하도록 변경하였습니다.

jeongmincha commented:
회의 마지막 시간에 말씀드렸었는데, DataType클래스에서 type명시를 할 때, 하나의 struct나 union도 DataType클래스의 인스턴스가 될 수 있기 때문에 그에 사용하기 위해서 STUCT랑 UNION을 추가하기 위함입니다

jeongmincha commented:
예를 들어서, 파서 룰 중에서 struct_union_type -> STRUCT | UNION이 되는데, 그 때, 
STRUCT { return STRUCT; } // 오른쪽의 STRUCT는 TypeSpecifier에 있는 STRUCT이고, 왼쪽은 파서에서 쓰이는 겁니다.
UNION { return UNION; }
이런 식으로 쓰려고 추가하였습니다

jeongmincha commented:
저렇게 한 후에, member_definition -> struct_union_type ID ID 등으로 룰이 갈 때,
DataType dt = new DataType($3); // 이름 명시
dt.SetType ($1); // 이 부분때문에 STRUCT UNION 추가가 필요한것
...
이런 식으로 구현을 할 것이기 때문에, 필요하다고 생각했습니다

namhyung commented:
아, 답글을 남긴다는 것이 깜빡했네요. 결론적으로 말하면 커밋 자체에는 문제가 없어보이는데, 굳이 지금 merge할 필요도 없어 보이네요. data type 및 table이 완성되면 그 때 같은 PR로 묶어서 보내주시기 바랍니다.

EverettDunamu committed:
common: modified type specifier enumeration type

- 기존에 VarType으로 typedef되어 있던 enumeration type의 이름을
TypeSpecifier로 변경하였습니다.

- 이 열거형에 STRUCT, UNION, USER_DEFINED, NOT_DEFINED 타입을 추가하였습니다.

- 이름을 변경한탓에 VarType을 사용하고 있던 소스들을 전부
TypeSpecifier를 사용하도록 변경하였습니다.
